gazelle/README.md

```diff
@@ -234,28 +234,35 @@ py_test(
 ```
 
 You can control the naming convention for test targets by adding a gazelle directive named
-`# gazelle:python_test_naming_convention`.  See the instructions in the section above that 
+`# gazelle:python_test_naming_convention`.  See the instructions in the section above that
 covers directives.
 
 ### Binaries
 
 When a `__main__.py` file is encountered, this indicates the entry point
-of a Python program.
+of a Python program. A `py_binary` target will be created, named `[package]_bin`.
 
-A `py_binary` target will be created, named `[package]_bin`.
+When no such entry point exists, Gazelle will look for a line like this in the top level in every module:
+
+```python
+if __name == "__main__":
+```
+
+Gazelle will create `py_binary` target will be created for every module with such line, with the target name
+being the same as module name.
```

Comment:
@linzhp, do you think it would be a good idea to add a directive to disable this behaviour? Some people may want to avoid creating such binaries in certain directories and I am wondering if enabling `py_binary` generation per `module` by default and then giving an option to turn it off would be a better way to introduce this feature.

Reply:
Maybe we can wait for such feature request comes up?